libr/core/carg.c

```diff
@@ -0,0 +1,255 @@
+		}
+        }
+	free (buf);
+}
+
+/* This functon display list of arg with some colors */
+
+R_API void r_core_print_func_args(RCore *core) {
+	RAnalOp *op = {0};
+	RListIter *iter;
+	bool color = r_config_get_i (core->config, "scr.color");
+	if (!core->anal) {
+		return;
+	}
+	if (!core->anal->reg) {
+		return;
+	}
+	const char *pc = r_reg_get_name (core->anal->reg, R_REG_NAME_PC);
+	ut64 cur_addr = r_reg_getv (core->anal->reg, pc);
+	op = r_core_anal_op (core, cur_addr);
+	if (!op) {
+		return;
+	}
+	if (op->type == R_ANAL_OP_TYPE_CALL) {
+		RAnalFunction *fcn;
+		RAnalFuncArg *arg;
+		int i;
+		int nargs = 0;
+		bool onstack = false;
+		const char *fcn_name = NULL;
+		ut64 pcv = op->jump;
+		if (pcv == UT64_MAX) {
+			pcv = op->ptr;
+		}
+		fcn = r_anal_get_fcn_at (core->anal, pcv, 0);
+		if (fcn) {
+			fcn_name = fcn->name;
+		} else {
+			if (core->flags) {
+				RFlagItem *item = r_flag_get_i (core->flags, pcv);
+				if (item) {
+					fcn_name = item->name;
+				}
+			}
+		}
+		RList *list = r_core_get_func_args (core, fcn_name);
+		if (color) {
+			r_cons_println (Color_BBLUE"------------------------------------"Color_RESET);
+		} else {
+			r_cons_println ("------------------------------------");
+		}
+		if (!r_list_empty (list)) {
+			int argcnt = 0;
+			r_list_foreach (list, iter, arg) {
+				if (arg->cc_source && !strncmp (arg->cc_source, "stack", 5)) {
+					onstack = true;
+				}
+				r_cons_printf (" arg [%d] - %s : ", argcnt, arg->name);
+				print_format_values (core, arg->fmt, onstack, arg->src);
+				argcnt++;
+			}
+		} else {
+			if (fcn) {
+				nargs = fcn->nargs;
+			}
+			if (nargs > 0) {
+				for (i = 0; i < nargs; i++) {
+					ut64 v = r_debug_arg_get (core->dbg, R_ANAL_CC_TYPE_STDCALL, i);
+					r_cons_printf (" arg [%d] - 0x%"PFMT64x, i, v);
+					r_cons_newline ();
+				}
+			} else {
+				r_cons_println ("arg [0] : void");
+			}
+		}
+		if (color) {
+			r_cons_println (Color_BBLUE"------------------------------------"Color_RESET);
+		} else {
+			r_cons_println ("------------------------------------");
+		}
+	}
+	r_anal_op_fini (op);
+}
+
+/* Returns a list of RAnalFuncArg */
+R_API RList *r_core_get_func_args(RCore *core, const char *fcn_name) {
+	if (!fcn_name) {
+		return NULL;
+	}
+	RList *list = r_list_new ();
+	char *key = resolve_fcn_name (core->anal, fcn_name);
+	if (!key) {
+		return NULL;
+	}
+	const char *sp = r_reg_get_name (core->anal->reg, R_REG_NAME_SP);
+	int nargs = r_anal_type_func_args_count (core->anal, key);
+	const char *cc = r_anal_type_func_cc (core->anal, key);
+	const char *src = r_anal_cc_arg (core->anal, cc, 1); // src of first argument
+	if (!cc) {
+		// unsupported calling convention
+		return NULL;
+	}
+	int i;
+	ut64 spv = r_reg_getv (core->anal->reg, sp);
+	ut64 s_width = (core->anal->bits == 64)? 8: 4;
+	if (src && !strcmp (src, "stack_rev")) {
+		for (i = nargs - 1; i >= 0; i--) {
+			RAnalFuncArg *arg = R_NEW0 (RAnalFuncArg);
+			set_fcn_args_info (arg, core->anal, key, cc, i);
```

Comment:
possible null deref

Reply:
if u are taking about anal being null , i have added a check in set_fcn_args_info ()

Reply:
im talking about arg being null, if R_NEW0 fails